src/main/java/address/browser/AddressBookBrowser.java

```diff
@@ -0,0 +1,273 @@
+        pages.stream().forEach(browserTab -> {
+                if (rangeOfIndexToLoad.isWithinRange(listOfContactsDisplayed.indexOf(browserTab.getPerson()))){
+                    listOfAlreadyLoadedPerson.add(browserTab.getPerson());
+                }
+            });
+        return listOfAlreadyLoadedPerson;
+    }
+
+    /**
+     * Unload profile pages that are unneeded from the cache of the browser.
+     * @param listOfContactsDisplayed A list of contacts displayed on the UI.
+     * @param rangeOfIndexToLoad A range of index of the listOfContactsDisplayed that are to be loaded into the browser
+     *                           cache.
+     */
+    private void unloadProfilePages(List<Person> listOfContactsDisplayed, Range rangeOfIndexToLoad) {
+
+        for (BrowserTab browserTab: pages) {
+            int personIndexInFilteredPerson = listOfContactsDisplayed.indexOf(browserTab.getPerson());
+
+            if (personIndexInFilteredPerson == -1) {
+                continue;
+            }
+
+            if (!rangeOfIndexToLoad.isWithinRange(personIndexInFilteredPerson)) {
+                browserTab.unloadProfilePage();
+            }
+        }
+    }
+
+    /**
+     * Selects the tab that is used to load the profile page of the person.
+     * @param person
+     */
+    private void selectTabAssignedForPerson(Person person) {
+        pages.stream().filter(browserTab
+                -> browserTab.getPerson().equals(person)).forEach(browserTab -> selectTab(pages.indexOf(browserTab)));
+    }
+
+    /**
+     * Loads profile pages into the browser.
+     * @param listOfPersonsToBeLoaded
+     */
+    private void loadProfilePages(ArrayList<Person> listOfPersonsToBeLoaded) {
+        for (int c = 0; c < listOfPersonsToBeLoaded.size(); c++){
+            Optional<BrowserTab> notAssignedTab = pages.stream().filter(page -> !page.isAssigned()).findFirst();
+            int indexOfTab = 0;
+
+            if(notAssignedTab.isPresent()){
+                indexOfTab = pages.indexOf(notAssignedTab.get());
+            }
+
+            pages.get(indexOfTab).loadProfilePage(listOfPersonsToBeLoaded.get(c));
+        }
+    }
+
+     /**
+     * Gets a list of person that are needed to be loaded to the browser.
+     * @param filteredPersons
+     * @param indexOfPerson
+     * @return
+     */
+    private ArrayList<Person> getListOfPersonsToBeLoaded(List<Person> filteredPersons, int indexOfPerson) {
+        ArrayList<Person> listOfPersonsToBeLoaded = new ArrayList<>();
+
+        for (int i = 0; i < BrowserManager.NUMBER_OF_PRELOADED_PAGE && i < filteredPersons.size(); i++){
+            listOfPersonsToBeLoaded.add(filteredPersons.get((indexOfPerson + i) % filteredPersons.size()));
+        }
+        return listOfPersonsToBeLoaded;
+    }
+
+    private List<Person> getFilteredPersons() {
+        return filteredModelPersons.stream()
+                .map(person -> (Person) person)
+                .collect(Collectors.toList());
```

Comment:
I think he is using this to downcast `ModelPerson` to `Person` probably because their equals methods are incompatible.
Btw YL if this is the case you need to `.map(Person::new)` because each object's true class and method behaviours still follow `ModelPerson`.

I will change them `ModelPerson` to a container class instead of a subclass of `Person` in another PR.